sqlsynthgen/make.py

```diff
@@ -412,39 +406,45 @@ def make_src_stats(
     Returns:
         The dictionary of src-stats.
     """
-    if schema_name:
-        engine = create_engine_with_search_path(dsn, schema_name)
-    else:
-        engine = create_engine(dsn, echo=False, future=True)
-
     use_smartnoise_sql = config.get("use-smartnoise-sql", True)
+    # SmartNoiseSQL doesn't support asyncio
+    use_asyncio = not use_smartnoise_sql
+    engine = create_db_engine(dsn, schema_name=schema_name, use_asyncio=use_asyncio)
+
     if use_smartnoise_sql:
+        logging.warning(
+            "SmartNoiseSQL does not support asyncio, so make-stats queries"
+            " will be run sequentially."
+        )
         dp_config = config.get("smartnoise-sql", {})
         snsql_metadata = {"": dp_config}
 
-        def execute_query(conn: Any, stat_data: Dict[str, Any]) -> Any:
+        async def execute_query(engine: Any, stat_data: Dict[str, Any]) -> Any:
             privacy = snsql.Privacy(
                 epsilon=stat_data["epsilon"], delta=stat_data["delta"]
             )
-            reader = snsql.from_connection(
-                conn.connection,
-                engine="postgres",
-                privacy=privacy,
-                metadata=snsql_metadata,
-            )
-            private_result = reader.execute(stat_data["query"])
+            with engine.connect() as conn:
+                reader = snsql.from_connection(
+                    conn.connection,
+                    engine="postgres",
+                    privacy=privacy,
+                    metadata=snsql_metadata,
+                )
+                private_result = reader.execute(stat_data["query"])
             # The first entry in the list names the columns, skip that.
             return private_result[1:]
 
     else:
 
-        def execute_query(conn: Any, stat_data: Dict[str, Any]) -> Any:
-            result = conn.execute(text(stat_data["query"])).fetchall()
+        async def execute_query(engine: Any, stat_data: Dict[str, Any]) -> Any:
```

Comment:
Suggest func comment and param rename:

```suggestion
        async def execute_query(engine: Any, query_block: Dict[str, Any]) -> Any:
            """Execute query using an asynchronous SQLAlchemy engine."""
```